docs/resources/samples/move-rows-across-tables.md

```diff
@@ -0,0 +1,256 @@
+---
+title: 'Move rows across tables in Office Scripts'
+description: 'Learn how to move rows across tables by saving filters, then processing and reapplying the filters.'
+ms.date: 03/02/2021
+localization_priority: Normal
+---
+
+# Move rows across tables by saving filters, then processing and reapplying the filters
+
+This script achieves the following:
+
+* Select rows from the source table where the value in a column is equal to _some value_.
+* Move all selected rows into a another (target) table on another worksheet.
+* Reapply the relevant filters on the source table.
+
+## Video
+
+[![Watch step-by-step video](../../images/move-rows-filters-vid.jpg)](https://youtu.be/_3t3Pk4i2L0 "Step-by-step video")
+
+![Screenshots of workbook before and after](../../images/table-filter-before-after.png)
+
+## Solution
+
+There are two script variants. The main difference is in the way the rows are selected.
+
+* In one script, the rows are selected by applying the table filter and reading the visible range. [Script](#move-rows-using-table-filter)
+* In the second, the rows are selected simply by reading the values and extracting the row values. [Script](#move-rows-using-range-values)
+
+## Sample Excel file
+
+Download the file <a href="input-table-filters.xlsx">input-table-filters.xlsx</a> used in this solution to try it out yourself!
+
+## Office Scripts
+
+The solution contains two Office Scripts:
+
+1. [Move Rows Using Table Filter](#move-rows-using-table-filter)
+1. [Move Rows Using Range Values](#move-rows-using-range-values)
+
+### Move rows using table filter
+
+```ts
+function main(workbook: ExcelScript.Workbook) {
+
+  // Update the table names, column name, and look-up value as needed.
+  const TargetTableName = 'Table1';
+  const SourceTableName = 'Table2';
+  const NameOfColumnToFilterOn = 'Category';
+  const ValueToFilterOn = 'Clothing';
+
+  // Get the table objects.
+  let targetTable = workbook.getTable(TargetTableName);
+  let sourceTable = workbook.getTable(SourceTableName);
+  // If you don't know the table names, you can fetch first table on a given worksheet name.
+  // let targetTable = workbook.getWorksheet('Sheet1').getTables()[0];
+  // let sourceTable = workbook.getWorksheet('Sheet2').getTables()[0];
+
+  
+  if (!targetTable || !sourceTable) {
+    console.log(`Required tables missing - Check that both source (${TargetTableName}) and target (${SourceTableName}) tables are present before running the script.`);
+    return;
+  }
+
+  // Save all of the filter criteria.
+  // Initialize with an empty object to hold the filter criteria.
+  const tableFilters = {};
+  // For each table column, collect the filter criteria.
+  sourceTable.getColumns().forEach((column) => {
+    let colFilterCriteria = column.getFilter().getCriteria();
+    if (colFilterCriteria) {
+      // If we don't remove these two keys, the API fails for some reason. So, remove these.
```

Comment:
We should know why this happens.

Reply:
Tracking with [feature](https://office.visualstudio.com/OC/_workitems/edit/4924924)

Reply:
The issue has been reported and is being addressed by Excel API team.